### PR TITLE
refactor(components): el-page-header icon

### PR DESCRIPTION
**Pull request by adaex**
Please make sure these boxes are checked before submitting your PR, thank you!

- [ ] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [ ] Make sure you are merging your commits to `dev` branch.
- [ ] Add some descriptions and refer to relative issues for your PR.

### CODE REVIEW

File: docs/en-US/component/page-header.md [hunk 1]
```diff
@@ -20,11 +20,11 @@ page-header/custom-icon
...
 ## Attributes
 
-| Attribute | Description | Type   | Accepted Values | Default      |
-| --------- | ----------- | ------ | --------------- | ------------ |
-| icon      | icon        | string | —               | el-icon-back |
-| title     | main title  | string | —               | Back         |
-| content   | content     | string | —               | —            |
+| Attribute | Description    | Type              | Accepted Values | Default |
+| --------- | -------------- | ----------------- | --------------- | ------- |
+| icon      | icon component | string/ Component | —               | Back    |
```
Comment (sxzz) [inline docs/en-US/component/page-header.md:25]:
```suggestion
| icon      | icon component | string / Component | —               | Back    |
```

File: packages/components/page-header/src/page-header.ts [hunk 2]
```diff
@@ -1,9 +1,10 @@
-import type { ExtractPropTypes } from 'vue'
+import { Back } from '@element-plus/icons'
+import type { ExtractPropTypes, PropType, Component } from 'vue'
 
 export const pageHeaderProps = {
   icon: {
-    type: String,
-    default: 'el-icon-back',
+    type: [String, Object] as PropType<string | Component>,
```
Comment (sxzz) [inline packages/components/page-header/src/page-header.ts:6]: Use `buildProp` please.

File: packages/components/page-header/src/page-header.vue [hunk 3]
```diff
@@ -3,7 +3,7 @@
     <div class="el-page-header__left" @click="handleClick">
       <div v-if="icon || $slots.icon" class="el-page-header__icon">
         <slot name="icon">
-          <i :class="icon"></i>
+          <el-icon> <component :is="icon" /> </el-icon>
```
Comment (sxzz) [inline packages/components/page-header/src/page-header.vue:6]:
```suggestion
          <el-icon><component :is="icon" /></el-icon>
```

File: packages/components/page-header/src/page-header.ts (hunk 2 above)
Comment (adaex) [reply packages/components/page-header/src/page-header.ts:6]: We can update it in the new PR

**Review by jw-foss — APPROVED** [review]

File: packages/components/page-header/src/page-header.ts (hunk 2 above)
Comment (JeremyWuuuuu) [reply packages/components/page-header/src/page-header.ts:6]: Let's make a follow up PR for this then.

**Pull request merged**